File: cpp/Maximum Subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        if(nums.size() == 1) return nums[0];
        int sum = 0, _max = nums[0];
        for(int item : nums){
            sum += item;
            _max = max(sum, _max);
            sum = max(sum, 0);
        }
        return _max;
    }
};

File: py/Maximum Subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        _sum = 0
        _max = nums[0]
        for item in nums:
            _sum += item
            if _sum > _max:
                _max = _sum
            if _sum < 0:
                _sum = 0
        return _max
